irohad/main/irohad.cpp

```diff
@@ -150,57 +150,72 @@ int main(int argc, char *argv[]) {
     return EXIT_FAILURE;
   }
 
-  // Check if genesis block path was specified
-  if (not FLAGS_genesis_block.empty()) {
-    // If it is so, read genesis block and store it to iroha storage
-    iroha::main::BlockLoader loader;
-    auto file = loader.loadFile(FLAGS_genesis_block);
-    auto block = loader.parseBlock(file.value());
-
-    // Check that provided genesis block file was correct
-    if (not block) {
-      // Abort execution if not
-      log->error("Failed to parse genesis block");
-      return EXIT_FAILURE;
+  /// if there are any blocks in blockstore, then true
+  bool blockstore = irohad.storage->getBlockQuery()->getTopBlockHeight() != 0;
+
+  /// genesis block file is specified as launch parameter
+  bool genesis = not FLAGS_genesis_block.empty();
+
+  /// overwrite ledger flag was set as launch parameter
+  bool overwrite = FLAGS_overwrite_ledger;
+
+  if (genesis) {  // genesis block file is specified
```

Comment:
Please put truth table from description before this if else. I believe this should help for those who will read this code